docs/sql-performance-tuning.md

```diff
@@ -428,3 +428,118 @@ You can control the details of how AQE works by providing your own cost evaluato
       <td>3.2.0</td>
     </tr>
   </table>
+
+## Storage Partition Join
+
+Storage Partition Join (SPJ) is an optimization technique in Spark SQL that makes use the existing storage layout to avoid the shuffle phase.
+
+This is a generalization of the concept of Bucket Joins, which is only applicable for [bucketed](sql-data-sources-load-save-functions.html#bucketing-sorting-and-partitioning) tables, to tables partitioned by functions registered in FunctionCatalog. Storage Partition Joins are currently supported for compatible V2 DataSources.
+
+The following SQL properties enable Storage Partition Join.
```

Comment:
nit: perhaps `The following SQL properties enable Storage Partition Join and various optimizations of it`?

Reply:
done, added 'in different join queries with various optimizations.'  (as some flags are about different scenarios)